test/functional-api/integration/callouts/close-state-callouts.it-spec.ts

```diff
@@ -0,0 +1,655 @@
+
+    // Assert
+    expect(data.state).toEqual(CalloutState.CLOSED);
+  });
+
+  test('Close callout that has been published', async () => {
+    // Act
+    const res = await createCalloutOnCollaboration(
+      entitiesId.hubCollaborationId,
+      calloutDisplayName,
+      calloutNameID
+    );
+    calloutId = res.body.data.createCalloutOnCollaboration.id;
+
+    await updateCalloutVisibility(calloutId, CalloutVisibility.PUBLISHED);
+
+    await updateCallout(
+      calloutId,
+      calloutDisplayName,
+      calloutNameID,
+      'calloutDescription update',
+      CalloutState.CLOSED,
+      CalloutType.COMMENTS
+    );
+    const aspectsData = await getDataPerHubCallout(entitiesId.hubId, calloutId);
+    const data = aspectsData.body.data.hub.collaboration.callouts[0];
+
+    // Assert
+    expect(data.state).toEqual(CalloutState.CLOSED);
+  });
+});
+
+describe('Callout - Close State - User Privileges Discussions', () => {
+  let hubCalloutId = '';
+  let challengeCalloutId = '';
+  let opportunityCalloutId = '';
+  const discussionCalloutNameId = 'suggestions';
+  let hubAspectId = '';
+
+  beforeAll(async () => {
+    const preconditions = async (calloutId: string) => {
+      await postCommentInCallout(calloutId, 'comment on discussion callout');
+      const resAspectonHub = await createAspectOnCallout(
+        calloutId,
+        'aspectDisplayName',
+        cardNameID
+      );
+      const aspectDataCreate = resAspectonHub.body.data.createAspectOnCallout;
+      hubAspectId = aspectDataCreate.id;
+      await updateCallout(
+        calloutId,
+        calloutDisplayName,
+        discussionCalloutNameId,
+        'calloutDescription update',
+        CalloutState.CLOSED,
+        CalloutType.COMMENTS
+      );
+    };
+
+    const hubCallout = await getHubCalloutByNameId(
+      entitiesId.hubId,
+      discussionCalloutNameId
+    );
+    hubCalloutId = hubCallout[0].id;
+    await preconditions(hubCalloutId);
+
+    const challengeCallout = await getChallengeCalloutByNameId(
+      entitiesId.hubId,
+      entitiesId.challengeId,
+      discussionCalloutNameId
+    );
+    challengeCalloutId = challengeCallout[0].id;
+    await preconditions(challengeCalloutId);
+
+    const opportunityCallout = await getOpportunityCalloutByNameId(
+      entitiesId.hubId,
+      entitiesId.opportunityId,
+      discussionCalloutNameId
+    );
+    opportunityCalloutId = opportunityCallout[0].id;
+    await preconditions(opportunityCalloutId);
+  });
+
+  afterAll(async () => {
+    await deleteCallout(opportunityCalloutId);
+    await deleteCallout(challengeCalloutId);
+    await deleteCallout(hubCalloutId);
+  });
+
+  describe('Discussion Callout - Close State ', () => {
+    describe('DDT Users sending messages to closed Hub discussion callout', () => {
+      // Arrange
+      test.each`
+        userRole
+        ${TestUser.HUB_ADMIN}
+        ${TestUser.HUB_MEMBER}
```

Comment:
other two use `QA_USER` here, is this a mistake?

Reply:
Not a mistake, different users have different privileges on different entities (hub / challenge / opportunity)